### ISSUE TITLE
Error reading Prometheus: Post "http://localhost:9090/api/v1/query": dial tcp 127.0.0.1:9090: connect: connection refused

### ISSUE THREAD

**Issue by amityadav9918** [grafana/grafana#46310]
**Steps to reproduce**:

**Actual Result**:

**Expected Result**

**Relevant WCAG Criteria:** [#.#.# WCAG Criterion](link to https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0)

**Environment**:
- Grafana version:
- Data source type & version:
- User OS & Browser:
- Others:

**Comment by Elfo404** [comment]
Closing this as it's lacking basic info. Please follow the issue template and we can reopen this.

**Comment by Geek-Joey** [comment]
I also meet the problem when i config the prometheus , the prometheus sercvice is normal , there is my configuration.
1.prometheus.yml
```yml
scrape_configs:
  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
  - job_name: "prometheus"
    static_configs:
      - targets: ["localhost:9090"]
# Push Gateway
  - job_name: "pushgateway"
    static_configs:
      - targets: ["localhost:9091"]
        labels:
          instance: pushgateway

  # Node Exporter
  - job_name: "node exporter"
    static_configs:
      - targets: ["localhost:9100"]

# Kafka Exporter
  - job_name: "kafka exporter"
    static_configs:
      - targets: ["localhost:9308"]
```
2. the /etc/grafana/grafana.ini , i remain the default configuration.

```shell
#################################### Paths ####################################
[paths]
# Path to where grafana can store temp files, sessions, and the sqlite3 db (if that is used)
data = /var/lib/grafana

# Temporary files in `data` directory older than given duration will be removed
;temp_data_lifetime = 24h

# Directory where grafana can store logs
logs = /var/log/grafana

# Directory where grafana will automatically scan and look for plugins
plugins = /var/lib/grafana/plugins

# folder that contains provisioning config files that grafana will apply on startup and while running.
provisioning = conf/provisioning

#################################### Server ####################################
[server]
# Protocol (http, https, h2, socket)
;protocol = http

# The ip address to bind to, empty will bind to all interfaces
;http_addr =

# The http port  to use
;http_port = 3000

# The public facing domain name used to access grafana from a browser
;domain = localhost

# Redirect to correct domain if host header does not match domain
# Prevents DNS rebinding attacks
;enforce_domain = false
# The full public facing url you use in browser, used for redirects and emails
# If you use reverse proxy and sub path specify full url (with sub path)
;root_url = %(protocol)s://%(domain)s:%(http_port)s/

# Serve Grafana from subpath specified in `root_url` setting. By default it is set to `false` for compatibility reasons.
;serve_from_sub_path = false

# Log web requests
;router_logging = false

# the path relative working path
;static_root_path = public

# enable gzip
;enable_gzip = false

# https certs & key file
;cert_file =
;cert_key =

# Unix socket path
;socket =

# CDN Url
;cdn_url =

# Sets the maximum time using a duration format (5s/5m/5ms) before timing out read of an incoming request and closing idle connections.
# `0` means there is no timeout for reading the request.
;read_timeout = 0
```
can you give some adviece ?

**Comment by DaliDalmas** [comment]
I am new to grafana and I am facing the same problem.

**Comment by SACHINJAKHAR** [comment]
Actually I'm new to Prometheus and Grafana and I want to start my Azure VM monitoring both server level and application level monitoring. But I'm not getting the proper way to setup this. I don't want to go with Azure managed Prometheus and Grafana.
Please share step by step format to setup this.
